### DISCUSSION TITLE
How to handle indeterminate states w/ react-hook-form

### DISCUSSION THREAD

**Discussion by bstro** [react-hook-form/react-hook-form#6958] [Q&A] [answered]
I am trying to build a form that consists of checkboxes. These checkboxes potentially represent multiple values across different objects, and thus will sometimes need an indeterminate state. 

Here's a screenshot of what I'm trying to build, and a data structure that represents what I'm talking about: 

![image](https://user-images.githubusercontent.com/627290/139991833-0a0c713a-f329-4d68-a75b-1bfa4a6327ee.png)

```
{
  id1: {
    val1: true,
    val2: false,
    val3: true
  },
  id2: {
    val1: false,
    val2: false,
    val3: true
  },
  id3: {
    val1: true,
    val2: false,
    val3: true
  }
}
```

Is react-hook-form suitable for managing this kind of state, where the inputs are an amalgam of values across multiple objects? 

Here's [a codesandbox with the above data](https://codesandbox.io/s/great-hill-ry0mq?file=/src/App.tsx), if anyone wants to give it a whirl. It's using the previous version of react-hook-form, (6.15.x) which my org's project is currently still using, as well as react-aria's [useCheckbox](https://react-spectrum.adobe.com/react-aria/useCheckbox.html).

Thanks a bunch!

**Comment by bluebill1049** [comment, marked as answer]
I think this can be build into a controlled component, in fact if you don't need validation then you probably don't need to use react hook form at all.

**Reply by bstro** [reply]
Makes sense to me, thanks! Probably overkill to bring in RHF for this tiny widget.
